sumologic/resource_sumologic_cloud_to_cloud_source_test.go

```diff
@@ -0,0 +1,168 @@
+					testAccCheckCloudToCloudSourceExists(cloudToCloudResourceName, &cloudToCloudSource),
+					resource.TestCheckResourceAttrSet(cloudToCloudResourceName, "id"),
+					testAccWaitCloudToCloudSource(),
+				),
+			},
+			{
+				Config: testAccSumologicCloudToCloudSourceConfig(cName, cDescription, cCategory, updatedConfigJSON),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudToCloudSourceExists(cloudToCloudResourceName, &cloudToCloudSource),
+					resource.TestCheckResourceAttrSet(cloudToCloudResourceName, "id"),
+				),
+			},
+		},
+	})
+}
+
+// wait function for waiting before perfroming an update to cloud-to-cloud source as updates are blocked when the source is in pending state
+func testAccWaitCloudToCloudSource() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		//lintignore:R018
+		time.Sleep(30 * time.Second)
+		return nil
+	}
+}
+
+func testAccCheckCloudToCloudSourceDestroy(s *terraform.State) error {
+
+	client := testAccProvider.Meta().(*Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "sumologic_cloud_to_cloud_source" {
+			continue
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("Cloud-to-Cloud Source destruction check: Cloud-to-Cloud Source ID is not set")
+		}
+		id, err := strconv.Atoi(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		collectorID, err := strconv.Atoi(rs.Primary.Attributes["collector_id"])
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		s, err := client.GetCloudToCloudSource(collectorID, id)
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		if s != nil {
+			return fmt.Errorf("Cloud-to-Cloud Source still exists")
+		}
+	}
+	return nil
+}
+func testAccCheckCloudToCloudSourceExists(n string, cloudToCloudSource *CloudToCloudSource) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("Cloud-to-Cloud Source ID is not set")
+		}
+		id, err := strconv.Atoi(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("Cloud-to-Cloud Source id should be int; got %s", rs.Primary.ID)
+		}
+		collectorID, err := strconv.Atoi(rs.Primary.Attributes["collector_id"])
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		c := testAccProvider.Meta().(*Client)
+		cloudToCloudSourceResp, err := c.GetCloudToCloudSource(collectorID, id)
+		if err != nil {
+			return err
+		}
+		*cloudToCloudSource = *cloudToCloudSourceResp
+		return nil
+	}
+}
+
+func testAccSumologicCloudToCloudSourceConfig(cName, cDescription, cCategory, sConfig string) string {
+	return fmt.Sprintf(`
+resource "sumologic_collector" "test" {
+	name = "%s"
+	description = "%s"
+	category = "%s"
+}
+
+resource "sumologic_cloud_to_cloud_source" "universal" {
```

Comment:
can we name this resource okta source instead of universal? this is what customers would do